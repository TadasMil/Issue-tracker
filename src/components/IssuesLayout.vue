<script setup lang="ts" name="IssuesLayout">
import type { IIssue } from '@/types'

import IssuesListEmpty from './IssuesListEmpty.vue'

const { issues } = defineProps<{
  issues: IIssue[]
}>()
</script>

<template>
  <div v-if="issues.length">
    <slot />
  </div>

  <template v-else>
    <IssuesListEmpty />
  </template>
</template>
