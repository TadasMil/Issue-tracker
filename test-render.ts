import type { RenderOptions } from '@testing-library/vue'
import BootstrapVue3 from 'bootstrap-vue-3'
import { render } from '@testing-library/vue'

const customRender = (ui: any, options?: RenderOptions) =>
  render(ui, {
    ...options,
    global: {
      ...options?.global,
      plugins: [BootstrapVue3, ...(options?.global?.plugins || [])],
    },
  })

export * from '@testing-library/vue'
export { customRender as render }
