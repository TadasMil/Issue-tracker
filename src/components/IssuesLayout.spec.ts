import { describe, it, expect } from 'vitest'
import { render } from '../../test-render'

import IssuesLayout from './IssuesLayout.vue'

describe('IssuesLayout', () => {
  it('should render issue slot given', () => {
    const { html } = render(IssuesLayout, {
      props: {
        issues: [
          {
            id: '1',
            title: 'Issue 1',
            description: 'Issue 1 description',
            status: 'OPEN',
          },
        ],
      },
      slots: {
        default: '<div>Issue</div>',
      },
    })

    expect(html()).toMatchSnapshot()
  })

  it('should render the default empty component', () => {
    const { html } = render(IssuesLayout, {
      props: {
        issues: [],
      },
    })

    expect(html()).toMatchSnapshot()
  })
})
