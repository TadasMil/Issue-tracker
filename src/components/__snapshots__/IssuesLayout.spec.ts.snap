// Vitest Snapshot v1

exports[`IssuesLayout > should render issue slot given 1`] = `
"<div>
  <div>Issue</div>
</div>"
`;

exports[`IssuesLayout > should render the default empty component 1`] = `
"<div class=\\"card text-center text-muted shadow-sm border-0\\">
  <!---->
  <!---->
  <div class=\\"card-body\\">
    <!---->
    <!---->
    <div class=\\"h-100 d-flex flex-column justify-content-center align-items-center\\">
      <p class=\\"card-text\\">
      <h3 class=\\"mb-1\\">No issues found</h3>
      </p>
      <p class=\\"card-text\\">
      <p class=\\"mb-1\\">Here you'll see issues</p>
      </p>
    </div>
  </div>
  <!---->
  <!---->
</div>"
`;
